core/jazz_ui/src/app/pages/environment-overview/env-overview-section.component.ts

```diff
@@ -251,9 +268,163 @@ popup(state){
-
 }
 
+  getSafeDetails() {
+    this.reqArnArray = [];
+    this.noData = false;
+    this.rolesList = [];
+    this.http.get(`/jazz/t-vault/safes/${this.safeName}`).subscribe(
+      (res) => {
+        if(Object.keys(res.data.roles).length === 0){
+          this.noData = true;
+          this.isSecretLoading = false;
+        }
+        else {
+        this.rolesObj = res.data.roles;
+        this.isSecretLoading = false;
+        this.rolesList.push(Object.keys(this.rolesObj))
+        this.rolesList[0].map((item)=>{
+            this.reqArnArray.push(this.rolesObj[item].arn)
+        })
+        if(this.reqArnArray.length > 1){
+          this.firstSafeRole = this.reqArnArray.splice(0,1)
+        } else  if(this.reqArnArray.length === 1){
+          this.firstSafeRole = this.reqArnArray[0];
+          this.reqArnArray = []
+        }
+       }
+      },
+      (error) => {
+        this.isError = true;
+        this.isSecretLoading = false;
+        let errMessage = this.toastmessage.errorMessage(error, 'updateSecret');
+        this.toast_pop('error', 'Oops!', errMessage);
+      }
+    )
+  }
+ 
+  addRoleInSafe() {
+    this.isSecretLoading = true;
+    let payload = {
+      "arn": this.roleValue,
+      "permission": "read"
+    }
+    this.http.post(`/jazz/t-vault/safes/${this.safeName}/role`, payload).subscribe(
+      res => {
+        this.isAddOrDelete = false;
+        let successMessage = this.toastmessage.successMessage(res,"updateSecret");
+        this.toast_pop('success',"",successMessage);
+        this.getSafeDetails();
+      },
+      (error) => {
+        this.getSafeDetails();
+        this.isAddOrDelete = false;
+        let errMessage = this.toastmessage.errorMessage(error, 'updateSecret');
+        this.toast_pop('error', 'Oops!', errMessage);
+      }
+    )
+  }
+
+  onCompleteClick () {
+    let payload = {  "arn":this.arnToBeDeleted }
+    this.isSecretLoading = true;
+    this.deleteRole = false;
+    this.showDisplay = false;
+    this.editEvents = true;
+    this.http.delete(`/jazz/t-vault/safes/${this.safeName}/role`, payload).subscribe(
+      res => {
+        let successMessage = this.toastmessage.successMessage(res,"updateSecret");
+        this.toast_pop('success',"",successMessage);
+        this.isSecretLoading = false;
+        this.getSafeDetails();
+
+      },
+      (err) => {
+        this.isSecretLoading = false;
+        let errMessage = this.toastmessage.errorMessage(err, 'updateSecret');
+        this.toast_pop('error', 'Oops!', errMessage);
+      }
+    )
+  }
+
+  outSidePopup() {
+    this.deleteRole = false;
+  }
+
+  onDeleteClick(val) {
+    this.deleteRole = true
+    this.arnToBeDeleted = '';
+    this.arnToBeDeleted = val
+    this.confirmationHeader = this.toastmessage.customMessage("acknowledgementHeader", "secretConfirmation");
+    this.confirmationText = this.toastmessage.customMessage( "deleteRole", "secretConfirmation");
+  }
+
+  ongrpNameChange(val) {
```

Comment:
nitpick: `onGrpNameChange`

Reply:
UI is loaded incorrectly in endpoint:  http://d2ud06w58eohho.cloudfront.net/
![image](https://user-images.githubusercontent.com/1162453/70078367-696a6400-15b7-11ea-8fae-61832f5e1408.png)

Reply:
@swati-sh Please address the above concern and reach out if you need assistance.

Reply:
addressed the comments.

Reply:
addressed the comments.